producer.go

```diff
@@ -1232,7 +1247,8 @@ func (self *TopicProducerMgr) PublishAsyncWithJsonExt(topic string, body []byte,
 		if pid < 0 {
 			return nil, errors.New("pub need partition id")
 		}
-		return PublishWithJsonExt(topic, strconv.Itoa(pid), body, ext.ToJson())
+		afterCompressed := self.compress(topic, body, ext)
```

Comment:
We can move this outside to avoid do compress again while retrying.

Reply:
done